### PR TITLE
Deprecate examples CLI

### PR DESCRIPTION
**Pull request by avishniakov**
## Describe changes
I added deprecation note for `example` CLI to warn users about upcoming changes. Details on new mechanism will follow later, for now it is just an announcement.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [x] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [x] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Other (add details above)

### CODE REVIEW

**Comment by avishniakov** [comment]
> @avishniakov could you add it [to the deprecations internal docs](https://www.notion.so/zenml/Deprecations-903bcaacb5c04954b9f6133b3340a312) as well? We try to keep track of what we're deprecating / how long things have been in that state etc.

Thanks for pointing, added 👍🏼

File: src/zenml/cli/example.py [hunk 1]
```diff
@@ -881,6 +881,12 @@ def run(
         shell_executable: Manually specify the path to the executable that
             runs .sh files.
     """
+    # Log deprecation warning
+    logger.warning(
+        "The `example` CLI is going to be deprecated soon after 01/08/2023 "
+        "and will be no longer maintained by the ZenML core team after that."
```
[the remark this answers is not in the document]
Comment (strickvl) [reply src/zenml/cli/example.py:887]: Sorry for the formatting (doing it in web UI) but the text suits better I think.

**Review by strickvl — CHANGES_REQUESTED** [review]
I think we'll also need to have these messages on all the other `zenml example...` CLI commands as well.

File: src/zenml/cli/example.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (avishniakov) [reply src/zenml/cli/example.py]: Thanks, sounds better - added in [f36d475](https://github.com/zenml-io/zenml/pull/1693/commits/f36d475d77c25eb024be6eec08c46052975f48df)

**Review by strickvl — CHANGES_REQUESTED** [review]
Can you add the message to the other CLI commands as well?

**Review by strickvl — APPROVED** [review]
🙌

**Review by strickvl — APPROVED** [review]
Approved, but @htahir1 do you think we also should remove the instructions to use the examples CLI command from all the `examples` READMEs? Or leave it since they're being removed anyway soon?

**Review by htahir1 — APPROVED** [review]
LGTM. If I was to nitpick I would indicate in the message to the user that they should manually clone the examples and follow the instructions in the example README but not neccessary

@strickvl  As they are being removed, i wouldnt really do the effort right now (as these are deprecations anyway, not removals)

**Pull request merged**